File: asgn3/QuestionTF.cpp
#include <iostream>
#include <string>
#include <cctype>
#include "Question.h"
#include "QuestionTF.h"

QuestionTF::QuestionTF() {
    std::string questionText;
    std::string answerText;
    correct = false;
}

QuestionTF::QuestionTF(std::string question, std::string answer) {
    questionText = question;
    answerText = answer;
    correct = false;
}

void QuestionTF::showQuestion() {
    std::cout << questionText << std::endl;
    return;
}

void QuestionTF::showAnswer() {
    std::cout << answerText << std::endl;
    return;
}

bool QuestionTF::checkAnswer(std::string answer) {
    char c = std::tolower(answer.c_str()[0]);
    char a = std::tolower(answerText.c_str()[0]);

    return c == a;
}

File: asgn5/QuestionTF.java
/**
 * An object for true/false type questions.
 * 
 * @author Victor Soriano Mendoza {@literal <soriano.victorm@student.clackamas.edu>}
 * @version 1.0
 */
public class QuestionTF extends Question {
	
	/**
	 * A constructor for validating and loading data.
	 * <p>
	 * Use this constructor when the data should be validated and loaded upon object creation. If validation fails,
	 * the data is not loaded.
	 * 
	 * @see QuestionTF#validate(String, int)
	 * @see QuestionTF#loadData(String)
	 * 
	 * @param line  A string containing question data
	 * @param lineNumber  An integer whose value is the line number of the question data
	 */
	public QuestionTF(String line, int lineNumber) {
		if (this.validate(line, lineNumber)) {
			this.loadData(line);
		}
	}
	//--------------------------------------------------------------------------------------------------------------------------
	/**
	 * Basic constructor
	 * <p>
	 * Use this constructor when validation and data loading should be handled separately.
	 */
	public QuestionTF() {}
	//--------------------------------------------------------------------------------------------------------------------------
	@Override
	public void showQuestion() {
		// TODO Auto-generated method stub
		StringBuilder sb = new StringBuilder(this.question);
		sb.append(" (T/F)?");
		System.out.println(sb);
	}
	//--------------------------------------------------------------------------------------------------------------------------
	@Override
	public boolean checkAnswer(String givenAnswer) {
		char correct = this.answer.toLowerCase().charAt(0);
		char given = givenAnswer.toLowerCase().charAt(0);
		return correct == given;
	}
	//--------------------------------------------------------------------------------------------------------------------------
	@Override
	public boolean validate(String line, int lineNumber) {
		QuestionTFValidation qv = new QuestionTFValidation();
		try {
			qv.validate(line);
			return true;
		} catch (Exception e) {
			this.lineErrorMessage(line, lineNumber, e.getMessage());
			return false;
		}
	}
	//--------------------------------------------------------------------------------------------------------------------------
	@Override
	public void loadData(String line) {
		String[] tokens = line.split("\\|");
		this.question = tokens[2];
		this.answer = tokens[3];
	}

}
